mpas_analysis/test/test_mpas_xarray.py

```diff
@@ -41,6 +41,15 @@ def test_subset_variables(self):
                                                              yearoffset=1850))
         self.assertEqual(ds.data_vars.keys(), varList)
 
+        with self.assertRaisesRegexp(AssertionError, 'Empty dataset is returned.'):
```

Comment:
@pwolfram, can you explain this usage of a `with` statement?  I'm only familiar with using `with` when you want to open a file (or create another object) but you want the resulting object to be deleted at the end of the scope of the `with` statement.  Is there a corresponding object here? Or does the `with` have a different purpose?